src/components/BookmarkButton/BookmarkButton.jsx

```diff
@@ -0,0 +1,37 @@
+import React from 'react'
+import {useLoading} from "../../hooks/useLoading";
+import {useAuth} from "../../contexts/Providers/AuthProvider/AuthProvider";
+import {useBookmarks} from "../../contexts/Providers/BookmarksProvider/BookmarksProvider";
+import {removeFromBookmarks, addToBookmarks} from "../../contexts/Providers/BookmarksProvider/helpers";
+
+
+function BookmarkButton({prod, classes}) {
+const {authState} = useAuth();
+const {loading:bookmarkLoading, setLoading:setBookmarkLoading} = useLoading();
+const {bookmarksState, bookmarksDispatch} = useBookmarks();
+const {bookmarks} = bookmarksState;
+
+
+const addBookmarkHandler= (product)=>{
+    if(!authState.isAuthenticated){
+      alert("Need to login first");
+      return;
+    }
+     const token=localStorage.getItem("token");
```

Comment:
could create a variable for token instead of taking it out of local storage everytime

Reply:
Made the change